### DISCUSSION TITLE
Information don't get update when using query builder, and don't throw a error.

### DISCUSSION THREAD

**Discussion by tavaresmatheus** [doctrine/doctrine2#10573] [Support Questions]
I'm trying to do a update on a array of entity properties, as i don't know the properties that are going to be update i've looped the array of things to be modified, like as a user i can change my email, and not my name so i can't set it manualy, just by looping it.
The problem is that using the query builder to update it, it simple don't update it, and dont throw any kind of error.
This is my code:

```
public function update(
        array $userInformation,
        string $userId
    ): array
    {
        // App\Entity\User
        foreach ($userInformation as $key => $value) {
            $queryBuilder = $this->getEntityManager()->createQueryBuilder();
            $queryBuilder->update('App\Entity\User')
                ->set($key, $value)
                ->where('id = :id')
                ->setParameter(':id', $userId);
        }

        $user = $this->findById($userId);
        return [
            'id' => $user->getId(),
            'name' => $user->getName(),
            'email' => $user->getEmail(),
            'roles' => $user->getRoles(),
        ];
    }
```

**Comment by dannyvankooten** [comment]
Came here via Google. Wanted to answer real quickly, despite this being very old.

From the docs on [QueryBuilder](https://www.doctrine-project.org/projects/doctrine-orm/en/3.5/reference/query-builder.html#the-querybuilder):

> The QueryBuilder is a builder object only - it has no means of actually executing the Query.

So you need to actually execute the query for the database to be updated. 

```php
$query = $queryBuilder->getQuery();
$query->getResult(); // executes the query
```
